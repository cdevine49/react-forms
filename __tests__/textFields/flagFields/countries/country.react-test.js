import React from 'react';
import Country from '../../../../src/components/textFields/flagFields/countries/country';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

describe("Country", () => {
  describe("Without Props", () => {
    test("renders empty flag", () => {
      const tree = renderer.create(
        <Country />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe("With Props", () => {
    test("offset", () => {
      const tree = renderer.create(
        <Country offset="-44px" />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    test("name", () => {
      const tree = renderer.create(
        <Country name="United States" />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    test("code", () => {
      const tree = renderer.create(
        <Country code="+1" />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    test("className", () => {
      const tree = renderer.create(
        <Country className="test" />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    test("onClick", () => {
      const onClick = jest.fn();
      const component = mount(
        <Country onClick={onClick} />
      );
      expect(onClick.mock.calls.length).toBe(0);
      component.find('button').simulate('click');
      expect(onClick.mock.calls.length).toBe(1);
    });
  });
});