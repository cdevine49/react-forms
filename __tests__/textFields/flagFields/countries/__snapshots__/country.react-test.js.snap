// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country With Props className 1`] = `
<li
  className="country test"
>
  <button
    onClick={[Function]}
  >
    <div
      className="flag"
      style={Object {}}
    />
  </button>
</li>
`;

exports[`Country With Props code 1`] = `
<li
  className="country"
>
  <button
    onClick={[Function]}
  >
    <div
      className="flag"
      style={Object {}}
    />
    <span
      className="code"
    >
      +1
    </span>
  </button>
</li>
`;

exports[`Country With Props name 1`] = `
<li
  className="country"
>
  <button
    onClick={[Function]}
  >
    <div
      className="flag"
      style={Object {}}
    />
    <span
      className="name"
    >
      United States
    </span>
  </button>
</li>
`;

exports[`Country With Props offset 1`] = `
<li
  className="country"
>
  <button
    onClick={[Function]}
  >
    <div
      className="flag"
      style={
        Object {
          "backgroundImage": "url(/images/flags.png)",
          "backgroundPositionX": "0px",
          "backgroundPositionY": "-44px",
        }
      }
    />
  </button>
</li>
`;

exports[`Country Without Props renders empty flag 1`] = `
<li
  className="country"
>
  <button
    onClick={[Function]}
  >
    <div
      className="flag"
      style={Object {}}
    />
  </button>
</li>
`;
